deployments/helm/nvidia-dra-driver-gpu/templates/_helpers.tpl

```diff
@@ -66,14 +66,19 @@ app.kubernetes.io/instance: {{ .Release.Name }}
 {{- end }}
 
 {{/*
-Selector labels
+Selector labels: meant to select just the pods of the corresponding
+deployment/daemonset/... That is, this is different per-component.
 */}}
 {{- define "nvidia-dra-driver-gpu.selectorLabels" -}}
-{{- if .Values.selectorLabelsOverride -}}
-{{ toYaml .Values.selectorLabelsOverride }}
-{{- else -}}
-{{ include "nvidia-dra-driver-gpu.templateLabels" . }}
```

Comment:
I don't think we need the template labels in the selector labels. 

The current logic will write something like `nvidia-dra-driver-gpu-component=controller` into the `matchesLabels` selector, and that is precise, and not conflicting (unless one wants it to conflict with something).

Reply:
Note that the `nvidia-dra-driver-gpu.templateLabels` helper above also pulls in `Values.selectorLabelsOverride`. Here we only include the `templateLabels` if the override is not set -- in which case the `templateLabels` will NOT include the selector labels. This does seem recursive / convoluted though.

Reply:
Thanks for looking deeply. Very appreciated.

I will review this more carefully tomorrow.

I want to understand use cases that we want to support, and then remove complexity that is not currently needed.

(this nifty setup will have to be documented better so that it can survive if it must survive!)

Based on what I understand _today_, I don't think that template labels and selector labels should be entangled.

I think template labels, as far as I understood, are just labels added to any and all component deployed by this Helm chart. That looks good, and keeps things more or less organized.

Selector labels, on the other hand, have a deep technical (and internal, for hat matter) meaning. I am not sure if we even want to let people override those.

Reply:
I went into history. https://github.com/NVIDIA/k8s-device-plugin/commit/4820ac13a2e34e1b8ea188563c2b71de720e3b0b#r157221509.

> Note that the nvidia-dra-driver-gpu.templateLabels helper above also pulls in Values.selectorLabelsOverride

I think it's better to change that then, here. Will do.

It's already a challenge for me to put into English words what a good distinction is between "common labels" and "template labels". I will work on that.

For me, it's easier to see a clear distinction between [common labels, template labels] and [selector labels]. I want to reinforce that distinction.

Reply:
Added a new commit with commentary declaring intent. Not perfect, but I think it's important for us to express the system that we want to use here.